File: cpp/alerts/torrent_deleted.cpp
#include "./torrent_deleted.h"

namespace ltrs {

InfoHashCpp torrent_deleted_alert_get_info_hashes(lt::torrent_deleted_alert* alert) {
    return info_hash_t_to_info_hash_cpp(alert->info_hashes);
}

} // namespace ltrs

File: src/ffi/alerts/torrent_deleted.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type torrent_deleted_alert = crate::ffi::ffi::torrent_deleted_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/torrent_deleted.h");
        type InfoHashCpp = crate::ffi::ffi::InfoHashCpp;
        unsafe fn torrent_deleted_alert_get_info_hashes(
            alert: *mut torrent_deleted_alert,
        ) -> InfoHashCpp;
    }
}
